tools/cli/go-whisk-cli/commands/action.go

```diff
@@ -410,11 +410,18 @@ func getExec(artifact string, kind string, isDocker bool, mainEntry string) (*wh
 
     if !isDocker || ext == ".zip" {
         code, err = readFile(artifact)
-        exec.Code = &code
 
-        if (ext == ".jar") {
-            exec.Jar = base64.StdEncoding.EncodeToString([]byte(code))
-            exec.Code = nil
+        if ext == ".zip" || ext == ".jar" {
+            // Base64 encode the file
+            code = base64.StdEncoding.EncodeToString([]byte(code))
+            if (ext == ".zip") {
+                exec.Code = &code
+            } else {
+                exec.Code = nil
```

Comment:
The way to tell would be to create a action with a .jar file and make sure the `code` field is not present in the package sent from the CLI.